tests/oper.rs

```diff
@@ -267,15 +268,15 @@ fn fold_and_sum() {
 }
 
 fn range_mat(m: Ix, n: Ix) -> Array2<f32> {
-    Array::linspace(0., (m * n - 1) as f32, m * n).into_shape((m, n)).unwrap()
+    Array::linspace(0., (m * n) as f32 - 1., m * n).into_shape((m, n)).unwrap()
 }
 
 fn range_mat64(m: Ix, n: Ix) -> Array2<f64> {
-    Array::linspace(0., (m * n - 1) as f64, m * n).into_shape((m, n)).unwrap()
+    Array::linspace(0., (m * n) as f64 - 1., m * n).into_shape((m, n)).unwrap()
```

Comment:
I was confused about why the - 1 moved here -- it's actually a the same result either way, right?

Reply:
`m` and `n` are of type `usize`, so if `m` or `n` is zero, then `m * n` is `0usize`. `0usize - 1` panics due to underflow. By casting the product (`m * n`) to `f64` before subtracting `1`, there is no underflow. This change makes `range_mat64` work for cases where `m` or `n` is zero.